### PR TITLE
Remove "Any" predicate filter

### PR DESCRIPTION
**Pull request by protonspring**
This is non-functional and untested.

I don't think this custom predicate filter does enough to justify it's existence.  It creates an unnecessary abstraction layer, but doesn't make the code any more readable.  The code is clear enough without it.

### CODE REVIEW

File: src/movepick.cpp [hunk 1]
```diff
@@ -225,7 +222,7 @@ Move MovePicker::next_move(bool skipQuiets) {
       /* fallthrough */
 
   case BAD_CAPTURE:
-      return select<Next>(Any);
+      return select<Next>([&](){ return true; });
```
Comment (mstembera) [inline src/movepick.cpp]: Can't these just be
```C++
return select<Next>([](){ return true; });
```
w/o the & ?

Comment (mstembera) [inline src/movepick.cpp]: Can't these just be 
```C++
return select<Next>([](){ return true; });
```
w/o the & ?

Comment (protonspring) [reply src/movepick.cpp]: Works on my machine, but not for the other filters.

**Comment by mcostalba** [comment]
Patch is good for me.

**Pull request merged**